tools/test/collect_coverage.sh

```diff
@@ -25,6 +25,9 @@
 # Script expects that it will be started in the execution root directory and
 # not in the test's runfiles directory.
 
+# Exit on errors (ex: lcov is not installed).
+set -e
```

Comment:
I'm not very confident that there aren't other errors produced by this script that don't require exiting. For this particular reason it's better to check if lcov is installed and exit if it's not.

Reply:
Good point. Though I imagine the only acceptable error in this script is invoking the test itself. We would still want to fail the build if `lcov` generation also failed, yes?

Reply:
There are 2 acceptable errors: invoking the test and not having any coverage data. The latter is also a consequence of `lcov` not being installed so the run would still fail.